ddl/ingest/engine_mgr.go

```diff
@@ -99,6 +101,18 @@ func (m *engineManager) Unregister(jobID, indexID int64) {
 	m.MemRoot.Release(StructSizeEngineInfo)
 }
 
+// ResetWorkers reset the writer count of the engineInfo because
+// the goroutines of backfill workers have been terminated.
+func (m *engineManager) ResetWorkers(jobID, indexID int64) {
+	ei, exist := m.Load(indexID)
+	if !exist {
+		return
+	}
+	m.MemRoot.Release(StructSizeWriterCtx * int64(ei.writerCount))
+	m.MemRoot.ReleaseWithTag(encodeEngineTag(jobID, indexID))
```

Comment:
Here release all mem use by engine and writer， but when start to handle next partition when regist engine and writer again, the engine releated  mem will not be count due to the exist of engine.

Reply:
Nope, here we don't release engine struct memory(specifically, `StructSizeEngineInfo`). When all the backfill worker exit, `engineCacheSize` should be clear from the memory root.